### PR TITLE
Lifecycle suite issues

### PR DESCRIPTION
**Pull request by edcdavid**
Changes in this PR:
- Drain command terminates critical services such as authentication (auth-openshift) with disconnects current container sessions -> before the drain, terminate all sessions as they might be interrupted 
- Sometimes the drain commands cannot complete and returns with error 1 -> if the drain command cannot complete, report a platform problem and fail ~~skip~~ the ~~testcase~~ suite. ~~To support this case, refactor the generic reel to provide the exit code in reelMatch~~
- Partner pod sessions are also left dangling before refresh -> add cleanup for all types of containers
- oc/exec Sessions process keep running even when the session monitoring goroutine has closed -> modify the Close goexpect callback to support cleaning the process
- Bugfix in the determination on which nodes to install debug pods in 
- Waiting after draining the node/test until all pods are up and running before refreshing the pod sessions

### CODE REVIEW

**Comment by greyerof** [comment]
Could you explain/elaborate in the PR description the concrete problem/issues in the lifecycle TS that you are addressing here, please?

**Comment by edcdavid** [comment]
> Could you explain/elaborate in the PR description the concrete problem/issues in the lifecycle TS that you are addressing here, please?

fixed

File: pkg/tnf/handlers/base/redhat/version.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (r *Release) ReelFirst() *reel.Step {
 }
 
 // ReelMatch determines whether the container is based on Red Hat technologies through pattern matching logic.
-func (r *Release) ReelMatch(pattern, _, _ string) *reel.Step {
+func (r *Release) ReelMatch(pattern, _, _ string, status int) *reel.Step {
```
Comment (jun-rh) [inline pkg/tnf/handlers/base/redhat/version.go:85]: _ if 'status' is not used

Comment (edcdavid) [reply pkg/tnf/handlers/base/redhat/version.go]: fixed

**Review by greyerof — APPROVED** [review]

**Review by jun-rh — APPROVED** [review]

File: pkg/config/config.go [hunk 2]
```diff
@@ -111,6 +111,7 @@ func getOcSession(pod, container, namespace string, timeout time.Duration, optio
 		gomega.Expect(err).To(gomega.BeNil())
 		// Set up a go routine which reads from the error channel
 		go func() {
+			log.Infof("start watching the session with container %s/%s", oc.GetPodName(), oc.GetPodContainerName())
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply pkg/config/config.go]: fixed

File: pkg/config/config.go [hunk 3]
```diff
@@ -202,25 +203,46 @@ func (env *TestEnvironment) LoadAndRefresh() {
...
-			node.Oc.Close()
 			autodiscover.DeleteDebugLabel(name)
 		}
 	}
+	env.NodesUnderTest = nil
+	env.Config.Nodes = nil
+	env.DebugContainers = nil
+}
+
+// Resets the environment during the drain test since all the connections are affected
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply pkg/config/config.go]: fixed

**Pull request merged**